### PR TITLE
Transfer Status Page

### LINKED ISSUE #116: Update Discord Link
https://discord.gg/BeFnJA5Kjb

### PR DESCRIPTION
**Pull request by jubalm**
- changed routing mechanism to get url parameter

Resolves #116

### CODE REVIEW

File: app/ts/components/TransactionPage/TransactionReceipt.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			<div>Fetching transaction receipt from chain...</div>
+		</div>
+	)
+}
+
+const QueryRefected = ({ message }: { message: string }) => {
+	return (
+		<div class='grid gap-2 border border-red-400/50 px-4 py-3 bg-red-200/10'>
+			<div class='font-bold text-lg'>Unable to load transaction receipt</div>
+			<div>You provider returned the following error:</div>
```
Comment (KillariDev) [inline app/ts/components/TransactionPage/TransactionReceipt.tsx:45]:
```suggestion
			<div>Error:</div>
```

File: app/ts/components/TransactionPage/TransactionReceipt.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			<div class='break-all p-3 text-sm bg-white/10 text-white/50'>{message}</div>
+		</div>
+	)
+}
+
+const QueryResolved = (props: { receipt: TransactionReceipt }) => {
+
+	if (props.receipt === null) return <div class='px-4 py-3 border border-red-500/50 text-red-500'>Unable to find transaction in chain!</div>
```
Comment (KillariDev) [inline app/ts/components/TransactionPage/TransactionReceipt.tsx:53]:
```suggestion
	if (props.receipt === null) return <div class='px-4 py-3 border border-red-500/50 text-red-500'>Unable to find transaction on chain!</div>
```

File: app/ts/components/TransactionPage/TransactionResponse.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { useEffect } from "preact/hooks"
+import { useTransactionResponse } from "../../store/transaction.js"
+import { TransactionResponse } from "../../types.js"
```
Comment (KillariDev) [inline app/ts/components/TransactionPage/TransactionResponse.tsx:3]:
```suggestion
import { useEffect } from 'preact/hooks'
import { useTransactionResponse } from '../../store/transaction.js'
import { TransactionResponse } from '../../types.js'
```

File: app/ts/components/TransactionPage/TransactionResponse.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+			props.onSuccess(txn.value.value)
+			return <QueryResolved transactionHash={props.transactionHash} transactionResponse={txn.value.value} />
+	}
+}
+
+const QueryRejected = ({ message }: { message: string }) => {
+	return (
+		<div class='grid gap-2 border border-red-400/50 px-4 py-3 bg-red-200/10'>
+			<div class='font-bold text-lg'>Unable to load transaction details</div>
+			<div>You provider returned the following error:</div>
```
Comment (KillariDev) [inline app/ts/components/TransactionPage/TransactionResponse.tsx:34]:
```suggestion
			<div>Error:</div>
```

File: app/ts/components/TransactionPage/TransactionResponse.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+			<div class='break-all p-3 text-sm bg-white/10 text-white/50'>{message}</div>
+		</div>
+	)
+}
+
+const QueryPending = () => {
+	return (
+		<div class='grid gap-2 grid-cols-[auto,1fr] items-center border border-white/20 px-4 py-3 bg-white/5'>
+			<svg width="1em" height="1em" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg" fill="none" class="animate-spin"><g fill="currentColor" fill-rule="evenodd" clip-rule="evenodd"><path d="M8 1.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zM0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8z" opacity=".2" /><path d="M7.25.75A.75.75 0 0 1 8 0a8 8 0 0 1 8 8 .75.75 0 0 1-1.5 0A6.5 6.5 0 0 0 8 1.5a.75.75 0 0 1-.75-.75z" /></g></svg>
```
Comment (KillariDev) [inline app/ts/components/TransactionPage/TransactionResponse.tsx:43]:
```suggestion
			<svg width='1em' height='1em' viewBox='0 0 16 16' xmlns='http://www.w3.org/2000/svg' fill='none' class='animate-spin'><g fill='currentColor' fill-rule='evenodd' clip-rule='evenodd'><path d='M8 1.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zM0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8z' opacity='.2' /><path d='M7.25.75A.75.75 0 0 1 8 0a8 8 0 0 1 8 8 .75.75 0 0 1-1.5 0A6.5 6.5 0 0 0 8 1.5a.75.75 0 0 1-.75-.75z' /></g></svg>
```

File: app/ts/components/TransactionPage/TransactionResponse.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+type TransactionDetailsProps = {
+	transactionHash: string
+	transactionResponse: TransactionResponse | null
+}
+
+const QueryResolved = (props: TransactionDetailsProps) => {
+
+	if (props.transactionResponse === null) return <div class='px-4 py-3 border border-red-500/50 text-red-500'>Unable to find transaction in chain!</div>
```
Comment (KillariDev) [inline app/ts/components/TransactionPage/TransactionResponse.tsx:56]:
```suggestion
	if (props.transactionResponse === null) return <div class='px-4 py-3 border border-red-500/50 text-red-500'>Unable to find transaction on chain!</div>
```

File: app/ts/components/TransactionPage/index.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+					</div>
+				</a>
+			</div>
+
+			<RecentTransfers />
+
+			<div class='pl-4 mb-4'>
+				<div class='text-white/30 text-sm mb-2'>Support</div>
+				<div class='p-4 text-sm bg-white/5 border border-white/10 text-white/50'>
+					Join our discord channel to get support from our active community members and stay up-to-date with the latest news, events, and announcements.
```
Comment (KillariDev) [inline app/ts/components/TransactionPage/index.tsx:91]: add link

File: app/ts/library/constants.ts [hunk 8]
```diff
@@ -135,3 +135,5 @@ export function getAssetMetadata(chainId: string, address?: string) {
 	if (match === undefined) throw new Error('Specified asset is unsupported')
 	return match
 }
+
+export const STORAGE_KEY_RECENTS = 'txns'
```
Comment (KillariDev) [inline app/ts/library/constants.ts:139]: this could have a more verbose name. Eg "sentTransactions"

File: app/ts/store/recent-transfers.ts [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import { signal, useSignalEffect } from '@preact/signals';
+import { useEffect } from 'preact/hooks';
+import { STORAGE_KEY_RECENTS } from '../library/constants.js';
```
Comment (KillariDev) [inline app/ts/store/recent-transfers.ts:3]:
```suggestion
import { signal, useSignalEffect } from '@preact/signals'
import { useEffect } from 'preact/hooks'
import { STORAGE_KEY_RECENTS } from '../library/constants.js'
```

File: app/ts/store/recent-transfers.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+type RecentTransaction = {
+	hash: string
+	date: number
+}
+
+const getSessionStorageCache = () => {
+	const valueFromStorage = sessionStorage.getItem(STORAGE_KEY_RECENTS);
+	return valueFromStorage ? JSON.parse(valueFromStorage) as RecentTransaction[] : [];
+};
```
Comment (KillariDev) [inline app/ts/store/recent-transfers.ts:13]:
```suggestion
const getSessionStorageCache = () => {
	const valueFromStorage = sessionStorage.getItem(STORAGE_KEY_RECENTS)
	return valueFromStorage ? JSON.parse(valueFromStorage) as RecentTransaction[] : []
}
```

File: app/ts/store/recent-transfers.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+		sessionStorage.setItem(STORAGE_KEY_RECENTS, JSON.stringify(uniqueTxns))
+	})
+
+	useEffect(() => {
+		window.addEventListener('storage', syncCacheChange)
+		return () => window.removeEventListener('storage', syncCacheChange)
+	}, [])
+
+	return { recentTxns, add }
+};
```
Comment (KillariDev) [inline app/ts/store/recent-transfers.ts:38]:
```suggestion
}
```

File: app/ts/store/transaction.tsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
+import { useSignal, useSignalEffect } from "@preact/signals"
+import { AsyncProperty, useAsyncState } from "../library/preact-utilities.js"
+import { TransactionReceipt, TransactionResponse } from "../types.js"
+import { useProviders } from "./provider.js"
```
Comment (KillariDev) [inline app/ts/store/transaction.tsx:4]:
```suggestion
import { useSignal, useSignalEffect } from '@preact/signals'
import { AsyncProperty, useAsyncState } from '../library/preact-utilities.js'
import { TransactionReceipt, TransactionResponse } from '../types.js'
import { useProviders } from './provider.js'
```

**Review by KillariDev — APPROVED** [review]

File: app/ts/store/transaction.tsx (hunk 12 above)
Comment (jubalm) [reply app/ts/store/transaction.tsx:4]: apologies, looks like prettier isn't working here and i didn't realize

File: app/ts/components/App.tsx [hunk 13]
```diff
@@ -3,13 +3,22 @@ import { TransferPage } from './TransferPage/index.js'
...
 import { SplashScreen } from './SplashScreen.js'
+import { TransactionPage } from './TransactionPage/index.js'
 
 export function App() {
 	return (
 		<SplashScreen>
 			<Router>
-				<Route path='#assets' component={AssetsPage} />
-				<Route path='/' component={TransferPage} />
+				<h1>invalid</h1>
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/App.tsx]: this is unintentional, removed

File: app/ts/components/TokenManager/index.tsx [hunk 14]
```diff
@@ -101,7 +101,6 @@ const TokenCard = ({ token, onSelect }: TokenCardProps) => {
 			<div class='relative'>
 				<img class='w-12 h-12 mb-2' src={`/img/${token.address}.svg`} />
 				<div class='font-bold'>{token.name}</div>
-				<div class='text-white/50'>Balance: 123.456</div>
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/TokenManager/index.tsx:104]: yes, moving balance on another PR

**Comment by jubalm** [comment]
formatting issues were corrected in [f93ef65](https://github.com/DarkFlorist/lunaria/pull/115/commits/f93ef653a3bcf2f6c721967a1fb464ca4548ee9f)

File: app/ts/library/constants.ts (hunk 8 above)
Comment (jubalm) [reply app/ts/library/constants.ts:139]: `txns` is only used as storage key, or did you mean `STORAGE_KEY_RECENTS`?

File: app/ts/components/TransactionPage/index.tsx (hunk 7 above)
Comment (jubalm) [reply app/ts/components/TransactionPage/index.tsx]: i was going to move it to another PR but would just be a small tweak so committed it here [0092596](https://github.com/DarkFlorist/lunaria/pull/115/commits/0092596661526506ef64efc06297c927655576df)

**Pull request merged**